script.js

```diff
@@ -11,3 +45,313 @@ function hideColumnContents(className) {
     }
   }
 }
+
+function adjustWidthOfColumns(firstColumnToBeAdjusted, secondColumnToBeAdjusted, headerOfFirstColumnToBeAdjusted, headerOfSecondColumnToBeAdjusted, firstColumnToBeAdjustedWidth, secondColumnToBeAdjustedWidth) {
+  for (const element of firstColumnToBeAdjusted) {
+    element.style.width = firstColumnToBeAdjustedWidth + 'px';
+    
+  }
+
+  headerOfFirstColumnToBeAdjusted.style.width = firstColumnToBeAdjustedWidth + 'px';
+
+  for (const element of secondColumnToBeAdjusted) {
+    element.style.width = secondColumnToBeAdjustedWidth + 'px';
+    
+  }
+
+  headerOfSecondColumnToBeAdjusted.style.width = secondColumnToBeAdjustedWidth + 'px';
+}
+
+document.addEventListener('DOMContentLoaded', function() {
+  const columnlineOne = document.querySelector('#columnline-one');
+  const columnlineTwo = document.querySelector('#columnline-two');
+  const columnlineThree = document.querySelector('#columnline-three');
+  const columnlineFour = document.querySelector('#columnline-four');
+  const columnlineFive = document.querySelector('#columnline-five');
+  var columnOne = document.getElementsByClassName('table__first-column');
+  var columnTwo = document.getElementsByClassName('table__second-column');
+  var columnThree = document.getElementsByClassName('table__third-column');
+  var columnFour = document.getElementsByClassName('table__fourth-column');
+  var headerOne = document.getElementById('table__first-header');
+  var headerTwo = document.getElementById('table__second-header');
+  var headerThree = document.getElementById('table__third-header');
+  var headerFour = document.getElementById('table__fourth-header');
+  var table = document.getElementById('table-id');
+
+  columnlineOne.onmousedown = function(event) {
+    function moveToMouse(windowCoordinatesX) {
+      const newColumnlineLeft = windowCoordinatesX;
+      const newColumnLeftIsInTable = newColumnlineLeft >= table.offsetLeft;
+
+      if (newColumnLeftIsInTable) {
+        columnlineOne.style.left = windowCoordinatesX + 'px';
+      }
+      else {
+        document.removeEventListener('mousemove', onMouseMove);
+      }
+    }
+
+    moveToMouse(event.clientX);
+
+    function onMouseMove(event) {
+      moveToMouse(event.clientX);
+    }
+
+    document.addEventListener('mousemove', onMouseMove);
+
+    columnlineOne.onmouseup = function() {
+      document.removeEventListener('mousemove', onMouseMove);
+    }
```

Comment:
Please refactor this code into a function that can be called for every columnline.

Reply:
So that you only have one line in the onmousedown funtion: the call of the refactored function